src/log_batch.rs

```diff
@@ -809,8 +819,10 @@ impl LogBatch {
     #[inline]
     pub(crate) fn prepare_write(&mut self, file_context: &LogFileContext) -> Result<()> {
         match self.buf_state {
-            BufState::Encoded(header_offset, entries_len) => {
-                LogItemBatch::prepare_write(&mut self.buf, file_context)?;
+            // `BufState::Sealed` means that `LogBatch` is under a repeated state of dumping.
+            BufState::Encoded(header_offset, entries_len)
+            | BufState::Sealed(header_offset, entries_len) => {
+                self.item_batch.prepare_write(&mut self.buf, file_context)?;
```

Comment:
> What if item batch is empty?

In this case, it will return `Ok(0)` directly when calling `engine.write(...)`.
https://github.com/tikv/raft-engine/blob/404e3fefaeeb4da6b7650268d500cfd3fbd29cae/src/engine.rs#L138-L141

Reply:
Better make the function more robust. Skip writing when empty should only be an optimization instead of being depended on.

Reply:
> Better make the function more robust. Skip writing when empty should only be an optimization instead of being depended on.

Got.